packages/paste-core/layout/grid/src/Grid.tsx

```diff
@@ -47,12 +51,14 @@ const Grid = React.forwardRef<HTMLDivElement, GridProps>(
               })
             : child
         ),
-      [children]
+      [children, vertical, equalColumnHeights, gutter]
     );
 
-    const gutterStyles = React.useMemo(() => getGutterStyles(gutter), [gutter]);
+    const gutterStyles = React.useMemo(() => getGutterStyles(gutter, vertical), [gutter, vertical]);
     const flexDirection = React.useMemo(() => getFlexDirection(vertical), [vertical]);
 
+    const finalStyles = {...gutterStyles, marginTop, marginBottom};
```

Comment:
nit: should this be something like `gridStyles`? I'm having flashbacks of finalVersion, finalFinalVersion, and so on.

Reply:
Yes, that sounds very reasonable. Thanks!